processes/catalog.py

```diff
@@ -25,71 +25,52 @@ def __init__(self, *args):
+        logger.info(f'Saved {len(self.records)} catalog records')
 
-        while True:
-            msgProps, _, msgBody = self.getMessageFromQueue(os.environ['OCLC_QUEUE'])
+    def process_catalog_messages(self, max_attempts: int=3):
+        for attempt in range(0, max_attempts):
+            while message := self.getMessageFromQueue(os.environ['OCLC_QUEUE']):
+                message_props, _, message_body = message
 
-            if msgProps is None:
-                if attempts <= max_attempts:
-                    waitTime = 60 * attempts
+                self.process_catalog_message(message_body)
+                self.acknowledgeMessageProcessed(message_props.delivery_tag)
 
-                    logger.info(f'Waiting {waitTime}s for OCLC catalog messages')
+            wait_time = 60 * (attempt + 1)
+            logger.info(f'Waiting {wait_time}s for OCLC catalog messages')
+            sleep(wait_time)
 
-                    sleep(waitTime)
-
-                    attempts += 1
-
-                    continue
-                else:
-                    logger.info('Exiting OCLC catalog process - no more messages.')
-                    break
-
-            attempts = 1
-
-            self.processCatalogQuery(msgBody)
-            self.acknowledgeMessageProcessed(msgProps.delivery_tag)
-
-    def processCatalogQuery(self, msgBody):
-        message = json.loads(msgBody)
-        oclcNo = message.get('oclcNo')
-        owiNo = message.get('owiNo')
+    def process_catalog_message(self, message_body):
+        message = json.loads(message_body)
+        oclc_number = message.get('oclcNo')
+        owi_number = message.get('owiNo')
         
-        catalogXML = self.oclcCatalogManager.query_catalog(oclcNo)
+        catalog_record = self.oclcCatalogManager.query_catalog(oclc_number)
 
-        if not catalogXML:
-            logger.warning(f'Unable to get catalog XML for OCLC number {oclcNo}')
+        if not catalog_record:
+            logger.warning(f'Unable to get catalog record for OCLC number {oclc_number}')
             return
         
-        self.parseCatalogRecord(catalogXML, oclcNo, owiNo)
-        logger.info(f'Processed OCLC catalog query for OCLC number {oclcNo}')
+        self.parse_catalog_record(catalog_record, oclc_number, owi_number)
+        logger.info(f'Processed OCLC catalog query for OCLC number {oclc_number}')
 
-    def parseCatalogRecord(self, catalogXML, oclcNo, owiNo):
+    def parse_catalog_record(self, catalog_record, oclc_number, owi_number):
         try:
-            parseMARC = etree.fromstring(catalogXML.encode('utf-8'))
-        except etree.XMLSyntaxError as e:
-            logger.error(f'OCLC catalog MARC xml is invalid for OCLC number: {oclcNo}')
-            logger.debug(e)
+            parsed_marc_xml = etree.fromstring(catalog_record.encode('utf-8'))
+        except Exception:
+            logger.exception(f'Unable to parse OCLC catalog MARC XML for OCLC number: {oclc_number}')
```

Comment:
Is it worth leaving in the debug level error so that we can see the way etree failed, if it fails? I suppose it's probably a rare case

Reply:
We will see the error in the exception log! However, I think if we are debugging locally we can add more debug logs. Since we set the log level as INFO for both QA and prod, these logs never get logged out.